csharp/src/Ice/Communicator.cs

```diff
@@ -201,6 +201,24 @@ public ObjectAdapter createObjectAdapterWithRouter(string name, RouterPrx router
         return instance.objectAdapterFactory().createObjectAdapter(name, router, serverAuthenticationOptions: null);
     }
 
+    /// <summary>
+    /// Gets the object adapter that is associated by default with new outgoing connections created by this
+    /// communicator. This method returns null unless you set a non-null default object adapter using
+    /// <see cref="setDefaultObjectAdapter" />.
+    /// </summary>
+    /// <returns>The object adapter associated by default with new outgoing connections.</returns>
+    /// <seealso cref="Connection.getAdapter" />
+    public ObjectAdapter? getDefaultObjectAdapter() => instance.outgoingConnectionFactory().getDefaultObjectAdapter();
```

Comment:
get/set DefaultObjectAdapter for consistency with createObjectAdapter above.

Unfortunately the Connection API is not consistent - we use get/set Adapter there.